polaris-react/src/components/Tabs/components/Tab/Tab.tsx

```diff
@@ -1,133 +1,451 @@
-import React, {useEffect, useRef} from 'react';
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+  useRef,
+  KeyboardEvent,
+  forwardRef,
+  RefObject,
+  MutableRefObject,
+  RefCallback,
+} from 'react';
+import {
+  InfoMinor,
+  DuplicateMinor,
+  Columns3Minor,
+  EditMinor,
+  DeleteMinor,
+  CaretDownMinor,
+} from '@shopify/polaris-icons';
 
 import {classNames} from '../../../../utilities/css';
-import {UnstyledLink} from '../../../UnstyledLink';
+import {useI18n} from '../../../../utilities/i18n';
 import {
   focusFirstFocusableNode,
   handleMouseUpByBlurring,
 } from '../../../../utilities/focus';
+import {useBreakpoints} from '../../../../utilities/breakpoints';
+import {UnstyledButton} from '../../../UnstyledButton';
+import {UnstyledLink} from '../../../UnstyledLink';
+import {Icon} from '../../../Icon';
+import {Popover} from '../../../Popover';
+import {ActionList} from '../../../ActionList';
+import {Modal} from '../../../Modal';
+import {Badge} from '../../../Badge';
+import {Inline} from '../../../Inline';
+import {Text} from '../../../Text';
+import type {TabPropsWithAddedMethods} from '../../types';
 import styles from '../../Tabs.scss';
 
-export interface TabProps {
-  id: string;
-  focused?: boolean;
-  siblingTabHasFocus?: boolean;
-  selected?: boolean;
-  panelID?: string;
-  children?: React.ReactNode;
-  url?: string;
-  measuring?: boolean;
-  accessibilityLabel?: string;
-  onClick?(id: string): void;
-}
+import {RenameModal, DuplicateModal} from './components';
 
-export function Tab({
-  id,
-  focused,
-  siblingTabHasFocus,
-  children,
-  onClick,
-  selected,
-  url,
-  panelID,
-  measuring,
-  accessibilityLabel,
-}: TabProps) {
-  const wasSelected = useRef(selected);
-  const panelFocused = useRef(false);
-  const node = useRef<HTMLLIElement | null>(null);
-
-  // A tab can start selected when it is moved from the disclosure dropdown
-  // into the main list, so we need to send focus from the tab to the panel
-  // on mount and update
-  useEffect(() => {
-    if (measuring) {
-      return;
-    }
+export const Tab = forwardRef(
+  (
+    {
+      content,
+      accessibilityLabel,
+      badge,
+      id,
+      panelID,
+      url,
+      onAction,
+      actions,
+      disabled,
+      isModalLoading,
+      icon,
+      siblingTabHasFocus,
+      measuring,
+      focused,
+      selected,
+      onToggleModal,
+      onTogglePopover,
+      viewNames,
+      tabIndexOverride,
+      onFocus,
+    }: TabPropsWithAddedMethods,
+    ref: RefObject<HTMLElement>,
+  ) => {
+    const i18n = useI18n();
+    const [popoverActive, setPopoverActive] = useState(false);
+    const [isRenameModalActive, setIsRenameModalActive] = useState(false);
+    const [isDeleteViewModalActive, setIsDeleteViewModalActive] =
+      useState(false);
+    const [isDuplicateModalActive, setIsDuplicateModalActive] = useState(false);
```

Comment:
I think we can simplify this:

```suggestion
   const [activeModalType, setActiveModalType] = useState<TabAction | null>(
      null,
    );
```